src/wasmv1_execution/abi/abis.rs

```diff
@@ -0,0 +1,318 @@
+pub fn abi_transfer_coins(
+    store_env: FunctionEnvMut<ABIEnv>,
+    arg_offset: i32,
+) -> Result<i32, WasmV1Error> {
+    handle_abi(
+        "transfer_coins",
+        store_env,
+        arg_offset,
+        |handler, req: proto::TransferCoinsRequest| -> Result<proto::Empty, WasmV1Error> {
+            let Some(proto::Address{address}) = req.to_address else {
+                return Err(WasmV1Error::RuntimeError("No address provided".into()));
+            };
+
+            let Some(proto::Amount{amount}) = req.raw_amount else {
+                return Err(WasmV1Error::RuntimeError("No coins provided".into()));
+            };
+
+            // Do not remove this. It could be used for gas_calibration in future.
+            // if cfg!(feature = "gas_calibration") {
+            //     let fname = format!("massa.{}:0", function_name!());
+            //     param_size_update(&env, &mut ctx, &fname, to_address.len(), true);
+            // }
+
+            handler
+                .interface
+                .transfer_coins(&address, amount)
+                .map_err(|err| {
+                    WasmV1Error::RuntimeError(format!("Transfer coins failed: {}", err))
+                })?;
+
+            Ok(proto::Empty {})
+        },
+    )
+}
+
+pub fn abi_generate_event(
+    store_env: FunctionEnvMut<ABIEnv>,
+    arg_offset: i32,
+) -> Result<i32, WasmV1Error> {
+    handle_abi(
+        "generate_event",
+        store_env,
+        arg_offset,
+        |_handler, req: proto::GenerateEventRequest| {
+            _handler
+                .interface
+                .generate_event(req.event)
+                .map_err(|err| {
+                    WasmV1Error::RuntimeError(format!("Failed to generate event: {}", err))
+                })?;
+
+            Ok(proto::Empty {})
+        },
+    )
+}
+
+/// Function designed to abort execution.
+pub fn abi_abort(store_env: FunctionEnvMut<ABIEnv>, arg_offset: i32) -> Result<i32, WasmV1Error> {
+    {
+        let mut store_env = store_env;
+        // get environment and interface
+        let env_mutex = store_env.data().clone();
+        let env_lock = env_mutex.lock();
+        let exec_env = env_lock.as_ref().ok_or_else(|| {
+            WasmV1Error::InstanciationError("ABIs cannot be called at initialization time.".into())
+        })?;
+        let interface = exec_env.get_interface();
+
+        // create handler
+        let mut handler = super::handler::ABIHandler {
+            store_env: &mut store_env,
+            exec_env,
+            interface,
+        };
+
+        // apply gas cost
+        let gas_cost = handler.get_gas_cost("abort");
+        if gas_cost > 0 {
+            handler.try_subtract_gas(gas_cost)?;
+        }
+
+        // read argument
+        let arg = handler.read_arg_raw(arg_offset)?;
+        let msg = "ABORT: ".to_string() + &String::from_utf8(arg).unwrap_or_default();
+        println!("{}", msg);
+
+        Err(WasmV1Error::RuntimeError(msg))
```

Comment:
Suggestions:
* no need to get the interface
* remove handler, no need to have the _raw functions in it
* remove gas cost
* use format! for better formatting
* use from_utf8_lossy for more robustness

```suggestion
        let mut store_env = store_env;
        let env_mutex = store_env.data().clone();
        let env_lock = env_mutex.lock();
        let exec_env = env_lock.as_ref().ok_or_else(|| {
            WasmV1Error::InstanciationError("ABIs cannot be called at initialization time.".into())
        })?;
        exec_env
            .read_buffer(&store_env, arg_offset)
            .map_err(|err| {
                WasmV1Error::RuntimeError(format!("Could not read ABI argument: {}", err))
            })?;
        let msg = format!("VM program triggered abort: {}", String::from_utf8_lossy(arg));
        debug!("{}", msg);
        Err(WasmV1Error::RuntimeError(msg))
```